R/geom-sf.R

```diff
@@ -197,11 +197,12 @@ geom_sf <- function(mapping = aes(), data = NULL, stat = "sf",
       mapping = mapping,
       stat = stat,
       position = position,
-      show.legend = if (is.character(show.legend)) TRUE else show.legend,
+      #show.legend = if (is.character(show.legend)) TRUE else show.legend,
+      show.legend = show.legend,
       inherit.aes = inherit.aes,
       params = list(
         na.rm = na.rm,
-        legend = if (is.character(show.legend)) show.legend else "polygon",
```

Comment:
Agree with @yutannihilation . @yutannihilation is insightful, very good!. This line indicates that a new parameter may be needed.